src/worker/worker.go

```diff
@@ -0,0 +1,74 @@
+package worker
+
+import (
+	"encoding/base64"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/f0o/turbo-pancake/common"
+	"github.com/f0o/turbo-pancake/utils"
+)
+
+//Worker : Actual Worker function that executes JobSpec
+func Worker(worker string) {
+	utils.PublishEvent(common.EventSpec{
+		ID:      worker,
+		Type:    "create",
+		Payload: "Worker Spawned",
+	})
+	for job := range common.JobQueue {
+		utils.Debug(job.ID, "Worker %s", worker)
+		if Filter(&job) {
+			cmd := exec.Command(os.Getenv("SHELL"), "-c", `INPUT=$(echo "$BASEINPUT" | __THREADER_INTERNAL=base64 `+os.Args[0]+`) eval $COMMAND`)
+			cmd.Env = append(os.Environ(),

```

Comment:
instead of appending the env array multiple times and than overwrite the current env you could use the 
os.Setenv("key","value") function.

Reply:
os.Setenv is not concurrency safe